common/src/main/java/com/dmdirc/parser/common/BaseParser.java

```diff
@@ -175,6 +175,23 @@ public CallbackManager getCallbackManager() {
         return callbackManager;
     }
 
+    /**
+     * Change the {@link CallbackManager} in use by this parser.
+     *
+     * This will shutdown the old CallbackManager before setting the new one.
+     *
+     * Subscribers are not carried across between callback managers.
+     *
+     * @param manager new CallbackManager to use (ignored if null)
+     */
+    protected void setCallbackManager(final CallbackManager manager) {
```

Comment:
I tried that in my first draft, but you can't use "this" when using super(), so I couldn't make it use the BaseParser handleCallbackError method - and you can't seem to add error handlers to the eventbus after-the-fact.